pkg/internal/checkup/checkup.go

```diff
@@ -259,134 +233,6 @@ func (c *Checkup) cleanupVMI() {
-		{Name: c.params.NetworkAttachmentDefinitionName, Namespace: c.namespace, MacRequest: c.params.TrafficGeneratorEastMacAddress.String()},
-		{Name: c.params.NetworkAttachmentDefinitionName, Namespace: c.namespace, MacRequest: c.params.TrafficGeneratorWestMacAddress.String()},
-	})
-	if err != nil {
-		return err
-	}
-
-	pciDevicesVarName, err := c.getPCIDevicesVarName(ctx)
-	if err != nil {
-		return err
-	}
-	trafficGeneratorPod := newTrafficGeneratorPod(c.params, secondaryNetworksRequest, pciDevicesVarName)
-
-	log.Printf("Creating traffic generator Pod %s..", ObjectFullName(c.namespace, trafficGeneratorPod.Name))
-	if c.params.Verbose {
-		trafficGeneratorPodJSON, _ := json.Marshal(trafficGeneratorPod)
-		log.Print(string(trafficGeneratorPodJSON))
-	}
-
-	c.trafficGeneratorPod, err = c.client.CreatePod(ctx, c.namespace, trafficGeneratorPod)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (c *Checkup) getPCIDevicesVarName(ctx context.Context) (string, error) {
-	netAttachDef, err := c.client.GetNetworkAttachmentDefinition(ctx, c.namespace, c.params.NetworkAttachmentDefinitionName)
-	if err != nil {
-		return "", err
-	}
-
-	const resourceNameAnnotationKey = "k8s.v1.cni.cncf.io/resourceName"
-	rawResourceName, exists := netAttachDef.Annotations[resourceNameAnnotationKey]
-	if !exists {
-		return "", fmt.Errorf("%q annotation is not found on the network-attachement-definition %q",
-			resourceNameAnnotationKey, ObjectFullName(c.namespace, c.params.NetworkAttachmentDefinitionName))
-	}
-
-	const (
-		pciDevicesVarPrefix    = "PCIDEVICE"
-		pciDevicesVarDelimiter = "_"
-	)
-
-	// build the PCI Device env var naming recipe
-	// https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin#pod-device-information
-	pciDevicesVarName := pciDevicesVarPrefix + pciDevicesVarDelimiter + rawResourceName
-	pciDevicesVarName = strings.ReplaceAll(pciDevicesVarName, ".", pciDevicesVarDelimiter)
-	pciDevicesVarName = strings.ReplaceAll(pciDevicesVarName, "/", pciDevicesVarDelimiter)
-	return strings.ToUpper(pciDevicesVarName), nil
-}
-
-func (c *Checkup) waitForPodRunningStatus(ctx context.Context, namespace, name string) error {
-	podFullName := ObjectFullName(c.namespace, name)
-	log.Printf("Waiting for Pod %s..", podFullName)
-	var updatedPod *k8scorev1.Pod
-
-	conditionFn := func(ctx context.Context) (bool, error) {
-		var err error
-		updatedPod, err = c.client.GetPod(ctx, namespace, name)
-		if err != nil {
-			return false, err
-		}
-		return pod.PodInRunningPhase(updatedPod), nil
-	}
-	const interval = time.Second * 5
-	if err := wait.PollImmediateUntilWithContext(ctx, interval, conditionFn); err != nil {
-		return fmt.Errorf("failed to wait for Pod '%s' to be in Running Phase: %v", podFullName, err)
-	}
-
-	log.Printf("Pod %s is Running", podFullName)
-	c.trafficGeneratorPod = updatedPod
-	return nil
-}
-
-func (c *Checkup) deletePod(ctx context.Context) error {
```

Comment:
shouldn't this be deleted in the former commit?

Reply:
No, because in case there is a setup failure - there is also a cleanup of the pod that uses this method.